src/plugins/Plugin.ts

```diff
@@ -42,11 +27,11 @@ export interface Plugin {
      * Manually record an event.
      * @param data Data that the plugin will use to create an event.
      */
-    record?(data: any): void;
+    record?<D extends unknown>(data: D): void;
 
     /**
      * Update the plugin.
-     * @param config Data that the plugin will use to update its config.
+     * @param updateWith Data that the plugin will use to update its config.
      */
-    update?(config: object): void;
+    update?<O extends unknown>(updateWith: O[]): void;
```

Comment:
issue: The array is specific to the DomEventPlugin -- the intent was that this could be any type. I suggest leaving this type unknown.
```suggestion
    update?<O extends unknown>(updateWith: O): void;
```

Reply:
Sure, would just need to update the implementation of it for the DOM plugin call.